pkg/tc/driver/cmdline/tc_cmdline_test.go

```diff
@@ -0,0 +1,415 @@
+		expectedCmdArgs = append(expectedCmdArgs, ingressQdisc.GenCmdLineArgs()...)
+		expectedCmdArgs = append(expectedCmdArgs, filterToDel.GenCmdLineArgs()...)
+
+		BeforeEach(func() {
+			fakeCmd = fakeExec.AddFakeCmd()
+		})
+
+		It("retuns no error when underlying command passes", func() {
+			fakeCmd.RunScript = append(fakeCmd.RunScript, newFakeAction(nil, nil, nil))
+
+			err := tcCmdLine.FilterDel(ingressQdisc, filterToDel)
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(fakeCmd.Argv).To(BeEquivalentTo(expectedCmdArgs))
+		})
+
+		It("retuns error when underlying command errors", func() {
+			fakeCmd.RunScript = append(fakeCmd.RunScript, func() ([]byte, []byte, error) {
+				return nil, nil, errors.New("test error!")
+			})
+
+			err := tcCmdLine.FilterDel(ingressQdisc, filterToDel)
+
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Context("FilterList", func() {
+		var fakeCmd *testingexec.FakeCmd
+		ingressQdisc := tctypes.NewIngressQDiscBuilder().Build()
+		expectedCmdArgs := []string{"tc", "-json", "filter", "list", "dev", fakeNetDev}
+		expectedCmdArgs = append(expectedCmdArgs, ingressQdisc.GenCmdLineArgs()...)
+		filterListOut := `[
+  {
+    "protocol": "ip",
+    "pref": 200,
+    "kind": "flower",
+    "chain": 0
+  },
+  {
+    "protocol": "ip",
+    "pref": 200,
+    "kind": "flower",
+    "chain": 0,
+    "options": {
+      "handle": 1,
+      "keys": {
+        "eth_type": "ipv4",
+		"ip_proto": "tcp",
+		"dst_port": 6666,
+        "dst_ip": "10.10.10.2/24"
+      },
+      "in_hw": true,
+      "in_hw_count": 1,
+      "actions": [
+        {
+          "order": 1,
+          "kind": "gact",
+          "control_action": {
+            "type": "pass"
+          },
+          "prob": {
+            "random_type": "none",
+            "control_action": {
+              "type": "pass"
+            },
+            "val": 0
+          },
+          "index": 2,
+          "ref": 1,
+          "bind": 1,
+          "used_hw_stats": [
+            "delayed"
+          ]
+        }
+      ]
+    }
+  }
+]`
+
+		BeforeEach(func() {
+			fakeCmd = fakeExec.AddFakeCmd()
+		})
+
+		It("retuns non empty flower filter without error when underlying command passes", func() {
+			fakeCmd.OutputScript = append(fakeCmd.OutputScript, newFakeAction([]byte(filterListOut), nil, nil))
+			expectedFilter := tctypes.NewFlowerFilterBuilder().
+				WithProtocol(tctypes.FilterProtocolIP).
+				WithPriority(200).
+				WithHandle(1).
+				WithChain(0).
+				WithMatchKeyDstIP("10.10.10.2/24").
+				WithMatchKeyDstPort(6666).
+				WithMatchKeyIPProto(string(policyrules.ProtocolTCP)).
+				WithAction(tctypes.NewGenericActionBuiler().WithPass().Build()).
+				Build()
+
+			filters, err := tcCmdLine.FilterList(ingressQdisc)
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(fakeCmd.Argv).To(BeEquivalentTo(expectedCmdArgs))
+			Expect(filters).To(HaveLen(1))
+			Expect(filters[0].Equals(expectedFilter)).To(BeTrue())
```

Comment:
is To(BeTrue()) part required?

Reply:
in this test i want to validate that FilterList outputs expected filters, so id like to compare it to an actual filter that is expected to be returned.

Expect call returns an Assertion which you need to check with a matcher so To(BeTrue()) is required

Reply:
Sorry, I misunderstood this construction. For some reason I thought Equals was part of the assertion check.
False positive, sorry for wasting your time